backend/core/src/main/java/org/sonarsource/sonarlint/core/BindingCandidatesFinder.java

```diff
@@ -55,16 +56,26 @@ public Set<String> findConfigScopesToBind(String connectionId, String projectKey
       return Set.of();
     }
 
-    Set<String> goodConfigScopeIdCandidates = new HashSet<>();
+    var goodConfigScopeCandidates = new HashSet<ConfigurationScope>();
 
     for (var scope : configScopeCandidates) {
       cancelMonitor.checkCanceled();
       if (checkIfScopeIsGoodCandidateForBinding(scope, connectionId, projectKey, cancelMonitor)) {
-        goodConfigScopeIdCandidates.add(scope.getId());
+        goodConfigScopeCandidates.add(scope);
       }
     }
 
-    return goodConfigScopeIdCandidates;
+    // if both a parent and a child configuration scope are candidates, preference should be given to the higher scope in the hierarchy
+    // we prefer to bind at the broadest possible scope
+    return filterOutLeafCandidates(goodConfigScopeCandidates);
+  }
+
+  private static Set<String> filterOutLeafCandidates(Set<ConfigurationScope> candidates) {
+    var candidateIds = candidates.stream().map(ConfigurationScope::getId).collect(Collectors.toSet());
+    return candidates.stream().filter(scope -> {
+      var parentId = scope.getParentId();
+      return parentId == null || !candidateIds.contains(parentId);
+    }).map(ConfigurationScope::getId).collect(Collectors.toSet());
```

Comment:
It's a minor thing but it would be nicer if we could avoid adding a leaf candidate to the collection in the first place rather than filtering them out later.

Reply:
But then we should be careful whether the child comes first which increases the check method complexity.

In this case, your version is preferable, yes

Reply:
Yes, I think it's simpler to remove it at the end, and performance-wise I don't think it makes a big difference